### ISSUE TITLE
deprecated pip feature causes build to fail in v.2.6.0 when attempting to create the wheel

### ISSUE THREAD

**Issue by anglyan** [Cantera/cantera#1486]
**Problem description**

When doing a source install using scons, recent versions of pip may fail to create a wheel due to deprecated flag `in-tree-build`.

**Steps to reproduce**

```
scons build
```

I believe the same error would pop up in any system with a version of pip equal or greater than 21.3.

**Behavior**

Here is the following error:

```
option --use-feature: invalid choice: 'in-tree-build' (choose from 'fast-deps', 'truststore', 'no-binary-enable-wheel-cache')
scons: *** [build/python/dist/Cantera-2.6.0-cp39-cp39-macosx_13_arm64.whl] Error 2
scons: building terminated because of errors.
```

Removing the flag in `cantera/python/SConstruct` solves the issue

**System information**

- Cantera version: v.2.6.0
- OS: mac, Ventura 13.3.1, m1 chip
- Python/MATLAB/other software versions: Python 3.9 version from homebrew

**Comment by bryanwweber** [comment]
Hi, thanks for reporting this! I believe this was fixed on `main` already. How did you obtain the source code for 2.6.0? Do you need to build with SCons or can you use the wheel from PyPI?

**Comment by bryanwweber** [comment]
See #1269 and #1272

**Comment by anglyan** [comment]
Oops, I missed #1272, this is clearly a duplicate.

I checked out the v.2.6.0 tag from the repo.

Since Python was enabled, SCons tried to create the wheel as part of the standard build process (I am also interested in the C++ interface as well).  Removing the offending flag worked for me, but I am not sure how it would work with earlier pip versions. I didn't check the wheel from PyPI, it would be interesting to compare now that I have it installed 'the hard way'.

**Comment by bryanwweber** [comment]
Thanks! If you're interested in C++, then you'll definitely need to be able to install from source. You can disable the Python interface with the `python_package=n` configuration option when you do `scons build`, if you want to. Also, as you saw, you can just delete that flag from SConscript with no problems! 😄

**Comment by ischoegl** [comment]
I believe this can be closed, given that the issue is already addressed in `main`?

**Comment by speth** [comment]
Yes, I'd say so. I don't think we're going to backport a fix for the 2.6.0 branch. I think that with the rapid pace at which breaking changes appear in setuptools, it is difficult to do anything more than ensure that our releases are compatible with a contemporaneous version of setuptools.
